### DISCUSSION TITLE
RTOS TCP connection to send data

### DISCUSSION THREAD

**Discussion by evercf** [cesanta/mongoose#2196] [Ask questions, get answers] [answered]
I found that the connection was successful, but the data was not sent when I used **mg_send**. When I used send function directly, the data was sent successfully.I read the content of **mg_send** function and found that **c->is_udp** is the key. When I connect to **tcp**, **c->is_udp** is 0, and send function is not used. Is there something I did not set

**Comment by scaprile** [comment, marked as answer]
https://github.com/cesanta/mongoose/discussions/2180#discussioncomment-5823822https://github.com/cesanta/mongoose/discussions/2180#discussioncomment-5823822

Please refrain from opening several discussions for the same subject, and follow the guidelines. (Response to your former question above, still applies)

You are not properly porting to your OS, that flag is only set when using UDP and you are using TCP. As the docs say, with TCP data is actually sent later when mg_mgr_poll() runs. If it is not sent it is because it is not finding your socket as having outstanding data and that is either because your port of your OS is not working or because you are not properly calling the manager. In any case, this is not the place to discuss that, as already said. So:
> 
> > You are not just _using_ Mongoose, you are _porting_ it to a different environment. You are on your own.
> 
> > Go to the `src` directory, look at the `arch_*` files, those are support for different environments, see what you can do for your case.
> 
> > Look at the `net_*` files, those are support for different TCP/IP stacks, see what you can do for your case.
> 
> > Look at `sock.c` there are some special cases handled there.
> 
> If you have a specific issue, open an issue, by all means fill the issue template and provide all necessary information. Attach the _**Minimum piece of code reproducing the issue**_

We don't run ThreadX so we can't reproduce your issue, all we can do is see your code and make suggestions

**Reply by evercf** [reply]
Ok, thank you
